Build failure with GHC 9.2.1

```
src/Generic/Random/Internal/BaseCase.hs:191:3: error:
    • Could not deduce (Alternative (IfM y Weighted Proxy))
        arising from the superclasses of an instance declaration
      from the context: (GBCSSum f g z e yf yg, GBCS f z yf e,
                         GBCS g z yg e, y ~ (yf ||? yg))
        bound by the instance declaration
        at src/Generic/Random/Internal/BaseCase.hs:(191,3)-(195,27)
    • In the instance declaration for ‘GBCS (f :+: g) z y e’
    |
191 |   ( GBCSSum f g z e yf yg
    |   ^^^^^^^^^^^^^^^^^^^^^^^...

src/Generic/Random/Internal/BaseCase.hs:239:3: error:
    • Could not deduce (Alternative (IfM y Weighted Proxy))
        arising from the superclasses of an instance declaration
      from the context: (GBCSProduct f g z e yf yg, GBCS f z yf e,
                         GBCS g z yg e, y ~ (yf &&? yg))
        bound by the instance declaration
        at src/Generic/Random/Internal/BaseCase.hs:(239,3)-(243,27)
    • In the instance declaration for ‘GBCS (f :*: g) z y e’
    |
239 |   ( GBCSProduct f g z e yf yg
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^...
cabal: Failed to build generic-random-1.5.0.0
```

--- a/src/Generic/Random/Internal/BaseCase.hs
+++ b/src/Generic/Random/Internal/BaseCase.hs
@@ -188,7 +188,9 @@ instance GBCS f z y e => GBCS (M1 i c f) z y e where
   gbcs y z = fmap M1 (gbcs y z)
 
 instance
-  ( GBCSSum f g z e yf yg
+  ( Alternative (IfM y Weighted Proxy)  -- logically redundant, but GHC isn't clever
+                                        -- enough to deduce; see #32
+  , GBCSSum f g z e yf yg
   , GBCS f z yf e
   , GBCS g z yg e
   , y ~ (yf ||? yg)
@@ -236,7 +238,9 @@ instance GBCSSumCompare f g z e 'GT where
   gbcsSumCompare _ _ _ g = fmap R1 g
 
 instance
-  ( GBCSProduct f g z e yf yg
+  ( Alternative (IfM y Weighted Proxy)  -- logically redundant, but GHC isn't clever
+                                        -- enough to deduce; see #32
+  , GBCSProduct f g z e yf yg
   , GBCS f z yf e
   , GBCS g z yg e
   , y ~ (yf &&? yg)

